### PR TITLE
add test data checks, fix predict ordering, update model.

### PR DESCRIPTION
**Pull request by anarkiwi**

### CODE REVIEW

File: networkml/algorithms/host_footprint.py [hunk 1]
```diff
@@ -110,6 +137,42 @@ def parse_args(raw_args=None):
...
+        df['role'] = df.filename.str.split('-').str[0]
+        # Drop unknown roles.
+        df = df[df['role'] != 'Unknown']
+        X = df.drop(['filename', 'role'], axis=1)
+        y = df.role
+        X = self.string_feature_check(X)
+        return (X, y)
+
+    def compare_test_data(self, model, scaler, label_encoder, test_data):
+        X_test, y_test = self._get_test_train_csv(test_data)
```
Comment (jspeed-meyers) [inline networkml/algorithms/host_footprint.py:155]: This looks good to me. I have a small suggestion. It would be more in line with sklearn documentation and perhaps more readable to persons accustomed to data science terminology to rename ```y_test``` to ```y_true```. Nice job, @anarkiwi

File: networkml/algorithms/host_footprint.py [hunk 2]
```diff
@@ -110,6 +137,42 @@ def parse_args(raw_args=None):
...
 
+    def _get_test_train_csv(self, path):
+        df = pd.read_csv(path)
+        df = self.reorder_drop_cols(df)
+        df = df.fillna(0)
+        # Split dataframe into X (the input features or predictors)
+        # and y (the target or outcome or dependent variable)
+        df['role'] = df.filename.str.split('-').str[0]
+        # Drop unknown roles.
+        df = df[df['role'] != 'Unknown']
```
Comment (jspeed-meyers) [inline networkml/algorithms/host_footprint.py:148]: A question: Do we want the code to explicitly drop unknown? What if a user wanted to include a class named "unknown"? The code would drop this class and it would require some digging by the user for the user to understand that.

File: networkml/trained_models/README.md [hunk 3]
```diff
@@ -2,25 +2,55 @@
...
-Currently one model is defined, depending upon the Host() featurizer. It was generated by:
+Currently one model is defined, depending upon the Host() featurizer.
+
+#### Training and predicting
+
+The model can be regenerated by:
 
 ~~~~
-networkml --first_stage=algorithm --final_stage=algorithm --trained_model=networkml/trained_models/host_footprint.json --label_encoder=networkml/trained_models/host_footprint_le.json --operation train --kfolds 5 ~/tmp/combined-host-ratios-041020.gz
+networkml --kfolds=5 --first_stage=algorithm --trained_model=networkml/trained_models/host_footprint.json --label_encoder=networkml/trained_models/host_footprint_le.json --scaler=networkml/trained_models/host_footprint_scaler.mod --operation train [--test_data=/tmp/test_host.csv] /tmp/train_host.csv
```
Comment (jspeed-meyers) [inline networkml/trained_models/README.md:29]: Does this command (given the current codebase) still include the unix time-related statistical features? I've been thinking that we should perhaps remove those statistical features from the code base. I worry that unsuspecting users will include those, get unrealistically good results, and not understand that the unrealistically good results are an artifact of the unix time-related statistical features.

**Comment by jspeed-meyers** [comment]
I like this! Wow. I have one meta-comment. I believe that the testing functionality is currently coupled with model training. In other words, a user can perform model assessment (getting accuracy, precision, recall, and F1 and also a confusion matrix) only by running the training command and also providing an optional switch aimed at unseen test data. Do I have that right? If so, I would suggest adding the capability to run the model assessment code without also running the model training code. For instance, a user might already have a model and simply want to assess its performance without rerunning model training. What do you think @cglewis @anarkiwi ?

**Comment by cglewis** [comment]
@jspeed-meyers that sounds like a good idea, please feel free to make a PR with your proposed enhancements.

File: networkml/trained_models/README.md (hunk 3 above)
Comment (anarkiwi) [reply networkml/trained_models/README.md:29]: No - class HostBase:

    CALC_COL_NAMES = (
        ('frame.len', 'frame_len'),
        ('frame.time_delta_displayed', 'time_delta'))

Only frame.len and frame.time_deta_displayed (interpacket arrival time). The canonical list is of course the featurizer CSV submitted to the training stage.

>>> import pandas as pd
>>> df = pd.read_csv('combined-host-ratios-041020.gz')
>>> cols = sorted(df.columns)
>>> [col for col in cols if 'time' in col]
['tshark_25q_time_delta_in', 'tshark_25q_time_delta_out', 'tshark_75q_time_delta_in', 'tshark_75q_time_delta_out', 'tshark_average_time_delta_in', 'tshark_average_time_delta_out', 'tshark_count_time_delta_in', 'tshark_count_time_delta_out', 'tshark_max_time_delta_in', 'tshark_max_time_delta_out', 'tshark_median_time_delta_in', 'tshark_median_time_delta_out', 'tshark_min_time_delta_in', 'tshark_min_time_delta_out', 'tshark_total_time_delta_in', 'tshark_total_time_delta_out', 'tshark_variance_time_delta_in', 'tshark_variance_time_delta_out'

File: networkml/algorithms/host_footprint.py (hunk 1 above)
Comment (anarkiwi) [reply networkml/algorithms/host_footprint.py:155]: Done

File: networkml/algorithms/host_footprint.py (hunk 2 above)
Comment (anarkiwi) [reply networkml/algorithms/host_footprint.py]: OK I added a flag train_unknown (default False). It logs a message by default saying it's dropping role Unknown to let the user know.

**Comment by anarkiwi** [comment]
"predict" bogon removed.
